File: 0014-longest-common-prefix/0014-longest-common-prefix.kt
class Solution {
    fun longestCommonPrefix(strs: Array<String>): String {
        var shortestStr = strs[0]
        var longestPrefix = ""
        for (i in strs) {
            shortestStr = if (shortestStr.count() < i.count()) shortestStr else i
        }

        var i = 0
        while (i < shortestStr.length) {
            if (strs.all { it.startsWith(shortestStr.slice(0..i))}) {
                longestPrefix = shortestStr.slice(0..i)
                i++
                continue
            }
            break
        }
        return longestPrefix
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let str = strs[0];
    let answer = '';

    for (let i = 0; i < str.length; i++) {
        let char = str[i];
        for (let n = 0; n < strs.length; n++) {
            if (strs[n][i] === char) continue;
            return answer;
        }
        answer += char;
    }
    
    return answer;
};